packages/store/addon/-private/identifiers/cache.ts

```diff
@@ -0,0 +1,426 @@
+import { DEBUG } from '@glimmer/env';
+import { assert } from '@ember/debug';
+import { Dict } from '../ts-interfaces/utils';
+import { ResourceIdentifierObject } from '../ts-interfaces/ember-data-json-api';
+import {
+  StableRecordIdentifier,
+  IS_IDENTIFIER,
+  DEBUG_CLIENT_ORIGINATED,
+  DEBUG_IDENTIFIER_BUCKET,
+  GenerationMethod,
+  UpdateMethod,
+  ForgetMethod,
+  ResetMethod,
+} from '../ts-interfaces/identifier';
+import coerceId from '../system/coerce-id';
+import uuidv4 from './utils/uuid-v4';
+import normalizeModelName from '../system/normalize-model-name';
+import isStableIdentifier from './is-stable-identifier';
+import isNonEmptyString from '../utils/is-non-empty-string';
+
+type IdentifierMap = Dict<string, StableRecordIdentifier>;
+type TypeMap = Dict<string, KeyOptions>;
+
+interface KeyOptions {
+  lid: IdentifierMap;
+  id: IdentifierMap;
+  _allIdentifiers: StableRecordIdentifier[];
+}
+
+let configuredForgetMethod: ForgetMethod;
+let configuredGenerationMethod: GenerationMethod;
+let configuredResetMethod: ResetMethod;
+let configuredUpdateMethod: UpdateMethod;
+
+export function setIdentifierGenerationMethod(method: GenerationMethod): void {
```

Comment:
How does this work with multiple apps/fastboot etc?

Reply:
generally speaking the generation method should be constant across instances, set once. The mechanics of this was discussed during the RFC process.